Decred Wallet/Features/Settings/SettingsController.swift

```diff
@@ -228,6 +228,7 @@ class SettingsController: UITableViewController  {
                 try AppDelegate.walletLoader.wallet?.delete(spendingPinOrPassword.utf8Bits)
                 DispatchQueue.main.async {
                     progressHud.dismiss()
+                    self.navigationMenuViewController()?.invalidateSyncTimer()
```

Comment:
I got a crash after deleting wallet but before this code was executed. The timer fired after wallet was deleted and attempted to call `wallet.GetBestBlockTimestamp` which led to a crash.
Call this function before the delete wallet code to prevent any possibility of a crash. Or wrap the timer callback in a try/catch statement.

Reply:
On further thought, do not call this invalidate function before the wallet is deleted. Only call it after wallet is successfully deleted. To that end, the code in line 231 should move to the `walletDeleted` function.